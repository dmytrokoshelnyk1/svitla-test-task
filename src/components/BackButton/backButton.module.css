.backButton {
  background: none;
  color: inherit;
  border: none;
  font: inherit;
  cursor: pointer;
  outline: inherit;
  padding: 5px;
}
.backButton:hover {
  background-color: rgba(183, 182, 182, 0.7);
  border-radius: 5%;
}
