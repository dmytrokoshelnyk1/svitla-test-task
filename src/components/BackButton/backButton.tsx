'use client'
import React from 'react'
import { useRouter } from 'next/navigation'
import styles from './backButton.module.css'

export const BackButton: React.FC = () => {
  const router = useRouter()

  return (
    <button
      className={styles.backButton}
      type="button"
      onClick={() => {
        router.back()
      }}
    >
      {'<- Go Back'}
    </button>
  )
}
