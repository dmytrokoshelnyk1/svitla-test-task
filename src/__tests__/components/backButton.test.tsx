// @ts-nocheck
import React from 'react'
import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'

import { BackButton } from '@/components/BackButton/backButton'
import { fireEvent } from '@testing-library/dom'
import { jest } from '@jest/globals'

const mockBackFn = jest.fn()
jest.mock('next/navigation', () => {
  return {
    __esModule: true,
    useRouter: () => ({
      push: jest.fn(),
      replace: jest.fn(),
      prefetch: jest.fn(),
      back: mockBackFn,
    }),
  }
})

describe('BackButton component tests', () => {
  it('Renders correctly initial component', async () => {
    render(<BackButton />)
    const btn = screen.getByText('<- Go Back')

    expect(btn).toBeInTheDocument()
  })

  it('Should handle rounter.push after click', () => {
    render(<BackButton />)

    const btn = screen.getByText('<- Go Back', { selector: 'button' })
    fireEvent.click(btn)

    expect(mockBackFn).toHaveBeenCalled()
  })

  it('renders correctly toMatchSnapshot', () => {
    const tree = renderer.create(<BackButton />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
