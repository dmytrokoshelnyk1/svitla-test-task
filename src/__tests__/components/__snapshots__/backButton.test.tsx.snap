// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton component tests renders correctly 1`] = `
<button
  className="backButton"
  onClick={[Function]}
  type="button"
>
  &lt;- Go Back
</button>
`;

exports[`BackButton component tests renders correctly toMatchSnapshot 1`] = `
<button
  className="backButton"
  onClick={[Function]}
  type="button"
>
  &lt;- Go Back
</button>
`;
